### DISCUSSION TITLE
Support for UE 4.25-Plus?

### DISCUSSION THREAD

**Discussion by ronan36880** [gildor2/UEViewer#253] [Q&A] [answered]
Hello there, I came across this amazing tool because I wanted to export some models from the game Five Nights at Freddy's: Security Breach.

After exploring the tool for a bit, I noticed that, when exporting certain rigs, it'd crash. And after digging around for a bit, I noticed that FNAF: SB seems to be using a custom version of 4.25 called 4.25-plus.

Yes, I know that on the forums there's a custom build, but I was wondering when official support would be added.

Would it be possible to have special support for this version? If not, how would I add it myself?

Thanks a lot for your time!

**Comment by gildor2** [comment]
Hi.
The main problem with UE4.25-Plus is a complexity of adding this engine version to the umodel's codebase. UE4 code assumes that engine version is just a NUMBER. If this would be a UE4.26 or something else - it's not a problem, I'd refer to it as GAME_UE4(26). But I can't use GAME_UE4(25+) or GAME_UE4(25-Plus), this simply won't let to compile the code. I didn't find a good solution for this yet, may be later.

**Reply by ronan36880** [reply]
Interesting, thanks a lot for the fast reply!

How should I proceed when it comes to extracting these models that are crashing the app?

Once again, I know that there is a build around on the forums that hopefully should address these issues, but I like it when i can see the source code of things and I wasn't able to find the patches used in that binary file that fix the issues.

**Reply by ronan36880** [reply]
Also, regarding the issue with the plus versions, couldn't that be a separate "boolean" setting the user can enable through a command line argument/checkbox in the gui?

**Comment by gildor2** [comment, marked as answer]
I've made an attempt to support UE4.25+. Uploaded the new built to gildor.org site. Please try it (I don't have any game files to test it myself).
As soon as I'll get a confirmation that it works, I'll push code to github and write an announcement.

**Reply by ronan36880** [reply]
@gildor2 thanks a lot for giving it a try! and I'm super sorry for not being able to reply any sooner.
I tested stuff around and it does appear to indeed work properly.

**Reply by gildor2** [reply]
Thanks for testing!
